[Serializer] improve documentation of default_context for the serializer (#17305)
symfony/symfony-docs: reference/configuration/framework.rst

This option allows to define an array of paths with files or directories where
the component will look for additional serialization files.

default_context
...............

**type**: ``array`` **default**: ``[]``

A map with default context options that will be used with each ``serialize`` and ``deserialize``
call. This can be used for example to set the json encoding behavior by setting ``json_encode_options``
to a `json_encode flags bitmask`_.

You can inspect the :ref:`serializer context builders <serializer-using-context-builders>`

> to discover the available settings.

Reviewer: ```suggestion
to discover the available settings.

.. versionadded:: 6.2

    The ``default_context`` parameter was introduced in Symfony 6.2.
```

Author: that is not correct. default_context has been added in symfony 5.4 but was not documented aparently. 
https://github.com/symfony/symfony/blob/5.4/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php#L1010

Author: the context builders seem to have been added in 6.1 https://github.com/symfony/symfony/tree/6.1/src/Symfony/Component/Serializer/Context

Reviewer: Oh ok, in this case I would vote for adding the default context parameter against `5.4` branch, and the yaml stuff against `6.2`

Author: right. i made https://github.com/symfony/symfony-docs/pull/17310

i will remove the changes from this MR, as they don't clash

Author: actually the part about context builders was only added later, in 6.1. i will rebase this on 6.2 when you merged upstream, to keep the reference.

best would be if all the context settings would be documented, but i don't have time to do that.

Reviewer: I merged and unmerged your other PR, thanks!

Author: thanks. rebased and solved the conflicts.

This option allows to define an array of paths with files or directories where
the component will look for additional serialization files.

default_context
...............

**type**: ``array`` **default**: ``[]``

A map with default context options that will be used with each ``serialize`` and ``deserialize``
call. This can be used for example to set the json encoding behavior by setting ``json_encode_options``
to a `json_encode flags bitmask`_.

You can inspect the :ref:`serializer context builders <serializer-using-context-builders>`
to discover the available settings.

> .. versionadded:: 6.2

Author: ```suggestion
.. versionadded:: 5.4
```

>     The ``default_context`` parameter was introduced in Symfony 6.2.

Author: ```suggestion
    The ``default_context`` parameter was introduced in Symfony 5.4.
```

call. This can be used for example to set the json encoding behavior by setting ``json_encode_options``
to a `json_encode flags bitmask`_.

You can inspect the :ref:`serializer context builders <serializer-using-context-builders>`
to discover the available settings.

.. versionadded:: 5.4

> 

Reviewer: Ro remove those messages, please rebase again, thanks

Author: thanks, rebased.
